vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/AlertSteps.java

```diff
@@ -57,7 +60,42 @@
     @When("I $alertAction alert with message which $comparisonRule `$message`")
     public void processAlert(Action alertAction, StringComparisonRule comparisonRule, String message)
     {
-        alertActions.processAlert(comparisonRule.createMatcher(message), alertAction);
+        String description = "Alert with message " + comparisonRule + " '" + message + "' should be present";
+        if (softAssert.assertTrue(description, alertActions.isAlertPresent()))
+        {
+            alertActions.processAlert(comparisonRule.createMatcher(message), alertAction);
+        }
+    }
+
+    /**
+     * Types text into the <b>alert</b> input field and then performs the specified action
+     * by either accepting the alert by clicking the 'OK' button or dismissing it by clicking the 'Cancel' button.
+     * <p>
+     * An <b>alert</b> is a small dialog box that appears on the display screen to provide information or to warn about
+     * a potentially damaging operation. To proceed, the user needs to confirm or dismiss the alert by clicking
+     * the 'OK' or 'Cancel' button, respectively.
+     * </p>
+     * <p>
+     * Note: This step is applicable only for alerts with a single input field.
+     * For authorization alerts, basic URL authorization should be used.
+     * For example, https://user:pass@domain.com.
+     * </p>
+     * <p>Actions performed at this step:</p>
+     * <ul>
+     * <li>Check if the alert is present on the page</li>
+     * <li>Type text into the alert input field</li>
+     * <li>Accept the alert</li>
+     * </ul>
+     * @param text String The text to be typed into the alert popup.
+     */
+    @When("I type text `$text` in alert and accept it")
+    public void typeTextInAlert(String text)
+    {
+        if (softAssert.assertTrue(ALERT_PRESENT, alertActions.isAlertPresent()))
```

Comment:
why not to reuse `doesAlertExist` ?

Reply:
this method uses BaseValidations.assertExpectedCondition method with deprecated code inside